scrape/scrape.go

```diff
@@ -1519,6 +1535,19 @@ loop:
 		if ok {
 			ref = ce.ref
 			lset = ce.lset
+
+			// Update metadata only if it changed in the current iteration.
+			if sl.appendMetadataToWAL {
+				sl.cache.metaMtx.Lock()
+				metaEntry, metaOk := sl.cache.metadata[yoloString([]byte(lset.Get(labels.MetricName)))]
+				if metaOk && metaEntry.lastIterChange == sl.cache.iter {
+					shouldAppendMetadata = true
```

Comment:
can we rename this to something like `metadataChanged`, to me `shouldAppendMetadata` sounds more like the name of a feature flag and seeing both like `if sl.appendMetadataToWAL && shouldAppendMetadata {` made me look for a minute to see why we had two variables that I initially though were the same thing